e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -163,14 +239,33 @@ public static async Task ServiceSendMethodAndVerifyResponseAsync(string deviceNa
             await serviceClient.CloseAsync().ConfigureAwait(false);
         }
 
-        public static async Task ServiceSendMethodAndVerifyResponseAsync(string deviceName, string methodName, string respJson, string reqJson, TimeSpan responseTimeout, ServiceClientTransportSettings serviceClientTransportSettings, TestLogger logger)
+        public static async Task ServiceSendMethodAndVerifyResponseAsync(string deviceId, string methodName, string respJson, string reqJson, TimeSpan responseTimeout, ServiceClientTransportSettings serviceClientTransportSettings, TestLogger logger)
```

Comment:
why have this and `public static async Task ServiceSendMethodAndVerifyResponseAsync(string deviceId, string methodName, string respJson, string reqJson, TestLogger logger)` both? Why not set optional params to default, and have just one method?

Reply:
Are you referring to the methods that take deviceId vs the methods that take deviceId and moduleId? I did try combining them like that, but the entire method ended being one big if/else block on the condition that the moduleId is present, so it didn't simplify much

Reply:
There are also some methods that take in `TimeSpan responseTimeout, ServiceClientTransportSettings serviceClientTransportSettings` as optional. Those seem like they can be combined. (line 224 vs line 242).